Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -248,6 +248,15 @@ extension PackageDescription.Package {
             dependencies = array.map { PackageDescription.Package.Dependency.fromJSON($0, baseURL: baseURL) }
         }
 
+        // Parse the compatible swift versions.
+        var compatibleSwiftVersions: [Int]? = nil
```

Comment:
If this is an empty array, there will be failure that the package's sources are incompatible with current tool. But I think empty array is worth having a special diagnosis. Will add that, thanks!